src/pages/organisation/openings/index.tsx

```diff
@@ -4,43 +4,76 @@ import MainWrapper from "@/wrappers/main"
+    const [data, setData] = useState<Opening[]>([]);
+    const [clickedOnOpening, setClickedOnOpening] = useState<boolean>(false)
+    const [openingClicked,setOpeningClicked]= useState<Opening>()
     const currentOrg = useSelector(currentOrgSelector);
-    console.log(currentOrg)
-    const getOpenings= async()=>{
-        const URL= `/org/${currentOrg.id}/openings/`
-        console.log(URL)
-        const res= await getHandler(URL)
-        console.log(res)
+    //change the name here maybe, its a string state
+    const [clickedOnOpeningId, setClickedOnOpeningId] = useState<string>('')
+
+    const getOpenings = async () => {
+        const URL = `/org/${currentOrg.id}/orgopenings`
+        const res = await getHandler(URL)
+        if (res.statusCode === 200) {
+           setData(res.data.openings);
+          } else {
+            if (res.data.message) Toaster.error(res.data.message, 'error_toaster');
+            else {
+              Toaster.error(SERVER_ERROR, 'error_toaster');
+            }
+          }
     }
-    useEffect(()=>{
+    useEffect(() => {
         getOpenings()
-    },[])
-    return(
-    <BaseWrapper title="Openings">
-        <OrgSidebar index={15}></OrgSidebar>
-        <MainWrapper>
-        <div className="w-full flex justify-between items-center">
-        <div className="w-fit text-6xl font-semibold dark:text-white font-primary">Openings</div>
-
-        <Plus
-                  onClick={() => setOpenModal(true)}
-                  size={42}
-                  className="flex-center rounded-full hover:bg-white p-2 transition-ease-300 cursor-pointer"
-                  weight="regular"
-                />
-        </div>
-        <NewOpening/>
-        </MainWrapper>
-    </BaseWrapper>)
-
-    
+    }, [])
+
+
+    return (
+        <BaseWrapper title="Openings">
+            {clickedOnOpening?<ViewOpening setClickedOnOpening={setClickedOnOpening} openingId={clickedOnOpeningId} openingClicked={openingClicked} setOpeningClicked={setOpeningClicked} data={data} setData={setData}/>:''}
+            <OrgSidebar index={15}></OrgSidebar>
+            <MainWrapper>
+                {openModal ? (<NewOpening setClickedOnNewOpening={setOpenModal} openings={data} setOpenings={setData}/>) : ''}
+                <div className="w-full flex justify-between items-center p-base_padding f ">
+
+                    <div className="w-fit text-6xl font-semibold dark:text-white font-primary ">Openings</div>
+                    {checkOrgAccess(ORG_MANAGER) ? (
+                        <Plus
+                            onClick={() => setOpenModal(!openModal)}
+                            size={42}
+                            className="flex-center rounded-full hover:bg-white p-2 transition-ease-300 cursor-pointer"
+                            weight="regular"
+                        />) : ''}
+
+
+                </div>
+
+                {
+                    data ? (data.map((data: any, index) => {
```

Comment:
openings.map(opening=>(
<OpeningCard opening={opening} setClickedOnOpening={setClickedOnOpening}/>
))

Reply:
Why passing all the attributes individually? just pass entire opening as prop